### DISCUSSION TITLE
Consequences of removing symfony/proxy-manager-bridge dependency for an application

### DISCUSSION THREAD

**Discussion by webignition** [symfony/symfony#51039] [Q&A] [answered]
I recently tried testing a Symfony-based application (initially Symfony 6.0, currently Symfony 6.3) under a development version of PHP 8.3 (commit hash [42e32a](https://github.com/php/php-src/commit/42e32a72b57a0dcf66bb1793fa4933859b758af0), possibly 8.3.0.beta1).

Composer dependency installation failed due to symfony/proxy-manager-bridge having an indirect dependency on something that does not like the PHP version being '8.3.0-dev'. This wasn't too surprising.

I created a draft PR for the same application with symfony/proxy-manager-bridge removed and all checks passed. I was somehow not expecting this and had assumed that symfony/proxy-manager-bridge was necessary.

I can see from the Symfony docs that Symfony 6.1 and lower required symfony/proxy-manager-bridge for lazy services but that this is no longer the case for Symfony 6.2 and higher. So symfony/proxy-manager-bridge is not needed for this aspect.

I can see from the commit history that symfony/proxy-manager-bridge was added to composer.json at the same time as doctrine components were added. Perhaps this is necessary for lazy entities? The application has only one entity with a relation (a ManyToOne association) and I can see from the ManyToOne attribute that the default fetch mode is lazy and this appears to continue to be working correctly.

So, finally to my questions: what might the consequences be of removing the symfony/proxy-manager-bridge dependency? Things seem to be working fine. Is that sufficient to indicate that the removal of the dependency is safe? Are there non-intuitive consequences that I am not aware of?

**Comment by HypeMC** [comment, marked as answer]
The `ProxyManagerBridge` has been deprecated since Symfony now provides lazy services out of the box (see #48484), so removing it shouldn't be a problem.
